types/react-navigation/index.d.ts

```diff
@@ -268,6 +268,7 @@ export interface NavigationInitAction {
 export interface NavigationResetAction {
   type?: 'Navigation/RESET',
   index: number,
+  key?: string | null,
```

Comment:
I think when using the `--strictNullChecks` flag you need `| null`. It wouldn't compile in my project without it. See https://www.typescriptlang.org/docs/handbook/release-notes/typescript-2-0.html